tests/validations/test_saved_query.py

```diff
@@ -87,6 +102,34 @@ def test_invalid_where_in_saved_query(  # noqa: D
     )
 
 
+def test_where_filter_validations_invalid_granularity(  # noqa: D
+    simple_semantic_manifest__with_primary_transforms: PydanticSemanticManifest,
+) -> None:
+    manifest = copy.deepcopy(simple_semantic_manifest__with_primary_transforms)
+
+    manifest.saved_queries = [
+        PydanticSavedQuery(
+            name="Example Saved Query",
+            description="Example description.",
+            query_params=PydanticSavedQueryQueryParams(
+                metrics=["bookings"],
+                group_by=["Dimension('booking__is_instant')"],
+                where=PydanticWhereFilterIntersection(
+                    where_filters=[
+                        PydanticWhereFilter(where_sql_template="{{ TimeDimension('metric_time', 'cool') }}"),
```

Comment:
Can we add more filters that are valid to make sure we don't warn incorrectly for those?
Also, I triggered a bug when I added the metric filter validations because I accidentally made these case sensitive for custom granularity names (that's been fixed since then). Can we add a test filter with an uppercase standard granularity like `DAY` to make sure the same thing doesn't happen here?